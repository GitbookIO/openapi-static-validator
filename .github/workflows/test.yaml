on: [push]
name: Test
jobs:
    test:
        name: Test
        runs-on: ubuntu-latest
        steps:
            - uses: actions/checkout@v3
            - uses: oven-sh/setup-bun@v1
            - run: bun install
            - run: bun run test
