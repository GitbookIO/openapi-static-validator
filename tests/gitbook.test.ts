import { expect, test } from 'bun:test';
import { validateRequest } from './gitbook.validate';

test('POST /spaces/1234/hive/token', () => {
    const result = validateRequest({
        path: '/spaces/1234/hive/token',
        method: 'post',
        headers: {
            'content-type': 'application/json',
        },
        query: {},
    });
    expect(result).toMatchSnapshot();
});
