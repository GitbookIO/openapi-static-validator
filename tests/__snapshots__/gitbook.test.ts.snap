// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`POST /spaces/1234/hive/token 1`] = `
{
  "headers": {
    "content-type": "application/json",
  },
  "method": "post",
  "operationId": "generateSpaceHiveReadAccessToken",
  "params": {},
  "path": "/spaces/1234/hive/token",
  "query": {},
}
`;
